azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -341,13 +339,25 @@ def _parse_plugin_settings(ext_handler, plugin_settings):
 
         name = ext_handler.name
         version = ext_handler.properties.version
-        settings = [x for x in plugin_settings \
-                    if getattrib(x, "name") == name and \
-                    getattrib(x, "version") == version]
 
-        if settings is None or len(settings) == 0:
+        ext_handler_plugin_settings = [x for x in plugin_settings if getattrib(x, "name") == name]
+        if ext_handler_plugin_settings is None or len(ext_handler_plugin_settings) == 0:
             return
 
+        settings = [x for x in ext_handler_plugin_settings if getattrib(x, "version") == version]
+        if len(settings) != len(ext_handler_plugin_settings):
```

Comment:
minor: these 2 statements may be a little more readable if all() is used:

`if all(getattrib(x, "version") == version x for x in ext_handler_plugin_settings):`

Reply:
Good point! I'm still going to keep it this way because incase this condition is not true, we would have to again walk the whole array to find the array for the settings with matching versions. It's not too big a deal but still extra computation that can be avoided.